### PR TITLE
Run cmake before running rustdoc

### LINKED ISSUE #2419: Shadow documentation CI workflow is broken
With #2404, Shadow's build system has significantly changed and `cargo doc` no longer completes successfully. [In the CI](https://github.com/shadow/docs/blob/cabc7551e474a5d9ba2b419ff2d10cbf2f7d2d11/.github/workflows/build.yaml#L26), we do essentially:

```bash
$ (cd shadow/src && cargo doc)
$ mv shadow/src/target/doc pages/docs/rust
```

We should either update Shadow so that `cargo doc` works, or update the CI workflow with a new command that builds the documentation.

```
[...]

The following warnings were emitted during compilation:

warning: In file included from ../../../src/lib/shadow-shim-helper-rs/shim_event.h:8,
warning:                  from ../../../src/lib/shadow-shim-helper-rs/ipc.h:6,
warning:                  from ../../../src/lib/shadow-shim-helper-rs/shim_shmem.h:16,
warning:                  from ../../../src/lib/shim/shim_syscall.h:7,
warning:                  from shadow_spinlock.c:7:
warning: ../../../src/main/host/syscall/kernel_types.h:27:10: fatal error: shd-config.h: No such file or directory
warning:    27 | #include "shd-config.h"
warning:       |          ^~~~~~~~~~~~~~
warning: compilation terminated.

error: failed to run custom build command for `shadow-shim-helper-rs v0.1.0 (/tmp/shadow/src/lib/shadow-shim-helper-rs)`

[...]

The following warnings were emitted during compilation:

warning: In file included from ../../src/lib/shadow-shim-helper-rs/shim_event.h:8,
warning:                  from ../../src/lib/shadow-shim-helper-rs/ipc.h:6,
warning:                  from ../../src/lib/shadow-shim-helper-rs/shim_shmem.h:16,
warning:                  from ../../src/main/host/host.h:22,
warning:                  from ../../src/main/core/worker.h:10,
warning:                  from ../../src/main/bindings/c/bindings.h:19,
warning:                  from core/logger/log_wrapper.c:4:
warning: ../../src/main/host/syscall/kernel_types.h:27:10: fatal error: shd-config.h: No such file or directory
warning:    27 | #include "shd-config.h"
warning:       |          ^~~~~~~~~~~~~~
warning: compilation terminated.

error: failed to run custom build command for `shadow-rs v2.2.0 (/tmp/shadow/src/main)`

[...]
```

### PR DESCRIPTION
**Pull request by sporksmith**
Fixes https://github.com/shadow/shadow/issues/2419

### CODE REVIEW

**Comment by sporksmith** [comment]
Not sure there's a good way to test before merging...?

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -23,6 +23,11 @@ jobs:
       with:
         path: shadow
         repository: shadow/shadow
+    # Run cmake to generate shd-config.h
+    - name: Run cmake
+      run: |
+        sudo apt-get install cmake
+        cmake -S . -B build
```
Comment (stevenengler) [inline .github/workflows/build.yaml]: I think you would need to cd into the `shadow` directory first?

**Review by stevenengler — APPROVED** [review]

**Comment by stevenengler** [comment]
> Not sure there's a good way to test before merging...?

I don't think so, but I think it's fine for this repo to just push to the main branch, and then delete the commit and overwrite the history if it didn't work. That's what I've been doing, and as long as we don't both try to work on this repo at the same time I think it should be fine.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (sporksmith) [reply .github/workflows/build.yaml]: thanks, fixed

**Pull request merged**